File: app/src/main/java/wangdaye/com/geometricweather/basic/model/option/provider/LocationProvider.java
package wangdaye.com.geometricweather.basic.model.option.provider;

import android.content.Context;

import androidx.annotation.Nullable;

import wangdaye.com.geometricweather.R;
import wangdaye.com.geometricweather.basic.model.option.utils.OptionUtils;

public enum LocationProvider {

    BAIDU("baidu"),
    BAIDU_IP("baidu_ip"),
    AMAP("amap"),
    NATIVE("native");

    private String providerId;

    LocationProvider(String id) {
        providerId = id;
    }

    public String getProviderId() {
        return providerId;
    }

    @Nullable
    public String getProviderName(Context context) {
        return OptionUtils.getNameByValue(
                context,
                providerId,
                R.array.location_services,
                R.array.location_service_values
        );
    }
}

File: app/src/main/java/org/breezyweather/common/basic/models/options/provider/LocationProvider.kt
package org.breezyweather.common.basic.models.options.provider

import android.content.Context
import org.breezyweather.R
import org.breezyweather.common.basic.models.options._basic.BaseEnum
import org.breezyweather.common.basic.models.options._basic.Utils

enum class LocationProvider(
    override val id: String
): BaseEnum {

    NATIVE("native"),
    BAIDU_IP("baidu_ip");

    companion object {

        fun getInstance(
            value: String
        ) = when (value) {
            "baidu_ip" -> BAIDU_IP
            else -> NATIVE
        }
    }

    override val valueArrayId = R.array.location_service_values
    override val nameArrayId = R.array.location_services

    override fun getName(context: Context) = Utils.getName(context, this)
}